planning/behavior_path_planner/autoware_behavior_path_planner_common/test/test_parking_departure_utils.cpp

```diff
@@ -0,0 +1,315 @@
+    path_with_lane_id.points.push_back(path_point_with_lane_id);
+  }
+  return path_with_lane_id;
+}
+
+TEST(BehaviorPathPlanningParkingDepartureUtil, calcFeasibleDecelDistance)
+{
+  using autoware::behavior_path_planner::utils::parking_departure::calcFeasibleDecelDistance;
+
+  auto data = std::make_shared<PlannerData>();
+  double velocity = 2.0;
+  double acceleration = 1.0;
+  double acceleration_limit = 2.0;
+  double jerk_limit = 1.0;
+  double target_velocity = 3.0;
+
+  auto odometry = std::make_shared<nav_msgs::msg::Odometry>();
+  odometry->pose.pose = autoware::test_utils::createPose(0.0, 0.0, 0.0, 0.0, 0.0, 0.0);
+  odometry->twist.twist.linear.x = velocity;
+  auto accel_with_covariance =
+    std::make_shared<autoware::behavior_path_planner::AccelWithCovarianceStamped>();
+  accel_with_covariance->accel.accel.linear.x = acceleration;
+  data->self_odometry = odometry;
+  data->self_acceleration = accel_with_covariance;
+  auto planner_data = std::static_pointer_cast<const PlannerData>(data);
+
+  // condition: current velocity is slower than target velocity
+  auto distance =
+    calcFeasibleDecelDistance(planner_data, acceleration_limit, jerk_limit, target_velocity);
+  ASSERT_TRUE(distance.has_value());
+  EXPECT_NEAR(distance.value(), 0.0, epsilon);
+
+  // condition: calculates deceleration distance
+  velocity = 5.0;
+  odometry->twist.twist.linear.x = velocity;
+  data->self_odometry = odometry;
+  planner_data = std::static_pointer_cast<const PlannerData>(data);
+  distance =
+    calcFeasibleDecelDistance(planner_data, acceleration_limit, jerk_limit, target_velocity);
+  ASSERT_TRUE(distance.has_value());
+  EXPECT_NEAR(distance.value(), 18.7730133, epsilon);
+
+  // condition: not valid condition
+  velocity = 0.3;
+  target_velocity = 0.0;
+  acceleration = -1.5;
+  jerk_limit = 0.25;
+  acceleration_limit = 2.0;
+  odometry->twist.twist.linear.x = velocity;
+  accel_with_covariance->accel.accel.linear.x = acceleration;
+  data->self_odometry = odometry;
+  data->self_acceleration = accel_with_covariance;
+  planner_data = std::static_pointer_cast<const PlannerData>(data);
+  distance =
+    calcFeasibleDecelDistance(planner_data, acceleration_limit, jerk_limit, target_velocity);
+  ASSERT_FALSE(distance.has_value());
+}
+
+TEST(BehaviorPathPlanningParkingDepartureUtil, modifyVelocityByDirection)
+{
+  using autoware::behavior_path_planner::utils::parking_departure::modifyVelocityByDirection;
+
+  const double velocity = 3.0;
+  const double target_velocity = 2.0;
+  const double acceleration = 2.0;
+
+  std::vector<PathWithLaneId> paths;
+  auto short_path = trajectory_to_path_with_lane_id(generateTrajectory<Trajectory>(1, 1.0));
+  auto long_path =
+    trajectory_to_path_with_lane_id(generateTrajectory<Trajectory>(10, 1.0, -velocity));
+  auto reverse_path =
+    trajectory_to_path_with_lane_id(generateTrajectory<Trajectory>(10, -1.0, velocity, -M_PI));
```

Comment:
question: the generated paths area not connected?

Reply:
No they are not. Is it better to connect. Would be nice to share your idea for a better test method!!!

Reply:
thanks! it' ok in this PR1